### PR TITLE
Align go version in GH actions

### PR DESCRIPTION
**Pull request by dsimansk**
As a follow-up to #2078 and partially to verify CI is green. I'd look into a bit more to have a reusable step with one place to set it ideally.

/cc @pierDipi

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

My question also is: do we even need go in the lint job?

**Comment by pierDipi** [comment]
Error doesn't really help

**Comment by ReToCode** [comment]
I partially remember that we had an issue with the golang `v2` action and bumping go. Not sure if its related, but mind trying to bump it to `actions/setup-go@v4`?

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -178,8 +178,8 @@ jobs:
       - name: Go Lint - knative-operator
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.50.1
-          args: --timeout=10m0s --verbose
+          version: v1.51.2
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply .github/workflows/validate.yaml:181]: To be more precise it's not more strict. They fixed recommended defaults actually. :)
Per commit: https://github.com/mgechev/revive/commit/aea6254fca65e15b6e0e0e9e2507af5470b2dc93

[the remark this answers is not in the document]
Comment (dsimansk) [reply .github/workflows/validate.yaml:181]: @pierDipi ^^

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**